pkg/cmd/origins/update/update.go

```diff
@@ -0,0 +1,142 @@
+package update
+
+import (
+  "strconv"
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/MakeNowJust/heredoc"
+
+	msg "github.com/aziontech/azion-cli/messages/origins"
+	api "github.com/aziontech/azion-cli/pkg/api/edge_applications"
+
+	"github.com/aziontech/azion-cli/pkg/cmdutil"
+	"github.com/aziontech/azion-cli/utils"
+  sdk "github.com/aziontech/azionapi-go-sdk/edgeapplications"
+	"github.com/spf13/cobra"
+)
+
+type Fields struct {
+  OriginKey            string
+	ApplicationID        int64
+	Name                 string 
+	OriginType           string 
+	Addresses            []string
+	OriginProtocolPolicy string 
+	HostHeader           string 
+	OriginPath           string 
+	HmacAuthentication   string
+	HmacRegionName       string 
+	HmacAccessKey        string 
+	HmacSecretKey        string 
+	Path                 string
+}
+
+func NewCmd(f *cmdutil.Factory) *cobra.Command {
+	fields := &Fields{}
+
+	cmd := &cobra.Command{
+		Use:           msg.OriginsUpdateUsage,
+		Short:         msg.OriginsUpdateShortDescription,
+		Long:          msg.OriginsUpdateLongDescription,
+		SilenceUsage:  true,
+		SilenceErrors: true,
+		Example: heredoc.Doc(`
+        $ azioncli origins update --application-id 1673635839 --origin-key "58755fef-e830-4ea4-b9e0-6481f1ef496d" --name "ffcafe222sdsdffdf" --addresses "httpbin.org" --host-header "asdf.safe" --origin-type "single_origin" --origin-protocol-policy "http" --origin-path "/requests" --hmac-authentication "false"
+        $ azioncli origins update --application-id 1673635839 --origin-key "58755fef-e830-4ea4-b9e0-6481f1ef496d" --name "drink coffe" --addresses "asdfg.asd" --host-header "host"
+        $ azioncli origins update --application-id 1673635839 --origin-key "58755fef-e830-4ea4-b9e0-6481f1ef496d" --in "update.json"
+        `),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			request := api.UpdateOriginsRequest{}
+			if cmd.Flags().Changed("in") {
+				var (
+					file *os.File
+					err  error
+				)
+				if fields.Path == "-" {
+					file = os.Stdin
+				} else {
+					file, err = os.Open(fields.Path)
+					if err != nil {
+						return fmt.Errorf("%w: %s", utils.ErrorOpeningFile, fields.Path)
+					}
+				}
+				err = cmdutil.UnmarshallJsonFromReader(file, &request)
+				if err != nil {
+					return utils.ErrorUnmarshalReader
+				}
+			} else {
+
+				if !cmd.Flags().Changed("application-id") || !cmd.Flags().Changed("origin-key") ||
```

Comment:
I think only application_id and origin-key are mandatory in update, right? OR you actually have to resend fields you don't want to update??